hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAHSv2ClientImpl.java

```diff
@@ -107,6 +110,45 @@ public void testGetAppAttemptReport() throws IOException, YarnException {
         isEqualTo("test original tracking url");
   }
 
+  @Test
+  public void testGetContainerByAppAttempt() throws IOException, YarnException {
+    final ApplicationId appId1 = ApplicationId.newInstance(0, 1);
+    final ApplicationId appId2 = ApplicationId.newInstance(0, 2);
+
+    final ApplicationAttemptId appAttemptId1 =
+            ApplicationAttemptId.newInstance(appId1, 1);
+    final ApplicationAttemptId appAttemptId2 =
+            ApplicationAttemptId.newInstance(appId2, 2);
+
+    final ContainerId containerId1 = ContainerId.newContainerId(appAttemptId1, 1);
+    final ContainerId containerId2 = ContainerId.newContainerId(appAttemptId2, 2);
+
+
+    Map<String, String> filter1 = new HashMap<>();
+    filter1.put("appattemptId", appAttemptId1.toString());
+    Map<String, String> filter2 = new HashMap<>();
+    filter2.put("appattemptId", appAttemptId2.toString());
+    when(spyTimelineReaderClient.getContainerEntities(appId1, "ALL", filter1,
+            0, null))
+            .thenReturn(createContainerEntities(containerId1));
+    when(spyTimelineReaderClient.getContainerEntities(appId2, "ALL", filter2,
+            0, null))
+            .thenReturn(createContainerEntities(containerId2));
+
+    when(spyTimelineReaderClient.getApplicationEntity(appId1, "ALL", null))
+            .thenReturn(createApplicationTimelineEntity(appId1, true,
+                    false));
+    when(spyTimelineReaderClient.getApplicationEntity(appId2, "ALL", null))
+            .thenReturn(createApplicationTimelineEntity(appId2, true,
+                    false));
+
+    List<ContainerReport> containerList1 = client.getContainers(appAttemptId1);
+    List<ContainerReport> containerList2 = client.getContainers(appAttemptId2);
+
+    assertThat(containerList1.size()).isEqualTo(1);
+    assertThat(containerList2.size()).isEqualTo(1);
```

Comment:
Can you please extend the checks to see whether the containerList contains the containers with the correct ids?